docs/dist/pre-commit.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+changed=$*
+if [[ "$changed" = "." ]];then
+    changed=$(find . -type f -name '*.php')
+fi
+if [[ -z "$changed" ]];then
+    changed=$(git diff --cached --name-only --diff-filter=ACM)
+fi
+if echo "$changed"|grep "\.php";then
+    CS_PHAR="docs/dist/phpcs.phar"
+    if [[ ! -e "$CS_PHAR" ]];then
+        curl -s https://squizlabs.github.io/PHP_CodeSniffer/phpcs.phar > $CS_PHAR
+    fi
+    if [[ ! -e "$CS_PHAR" ]];then
+        echo "PHP_CodeSniffer not available"
```

Comment:
This will only happen if directory `docs/dist` is not writable: line 13 creates the file, no matter whether curl succeeds or not.